packages/clerk-js/src/core/resources/SignUp.ts

```diff
@@ -207,7 +207,21 @@ export class SignUp extends BaseResource implements SignUpResource {
       clerkVerifyWeb3WalletCalledBeforeCreate('SignUp');
     }
 
-    const signature = await generateSignature({ identifier, nonce, provider });
+    let signature: string;
+    try {
+      signature = await generateSignature({ identifier, nonce, provider });
+    } catch (err) {
+      // There is a chance that as a first time visitor when you try to setup and use the
+      // Coinbase Wallet from scratch in order to authenticate, the initial generate
+      // signature request to be rejected. For this reason we retry the request once more
+      // in order for the flow to be able to be completed successfully.
+      if (provider === 'coinbase_wallet' && err.message === 'Request rejected') {
```

Comment:
Is there anything else we can check against ? Maybe a code ? Does this come back from the coinbase sdk ?

Reply:
We can also check the error type, but that means we will have to import the type from the coinbase sdk globally and i believe lose our lazy loading ability

Reply:
Replaced with an error code check instead of the message